File: tmp/src/cazzar/mods/jukeboxreloaded/gui/GuiHandler.java
package cazzar.mods.jukeboxreloaded.gui;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;
import cazzar.mods.jukeboxreloaded.blocks.TileJukeBox;
import cpw.mods.fml.common.network.IGuiHandler;

public class GuiHandler implements IGuiHandler {
    public static final int JUKEBOX = 0;

    @Override
    public Object getClientGuiElement(int ID, EntityPlayer player, World world,
            int x, int y, int z)
    {
        final TileEntity tile = world.getBlockTileEntity(x, y, z);

        switch (ID)
        {
            case JUKEBOX:
                if (!(tile instanceof TileJukeBox)) return null;
                return new GUIJukeBox(player, (TileJukeBox) tile);
            default:
                return null;
        }
    }

    @Override
    public Object getServerGuiElement(int ID, EntityPlayer player, World world,
            int x, int y, int z)
    {
        final TileEntity tile = world.getBlockTileEntity(x, y, z);

        switch (ID)
        {
            case JUKEBOX:
                if (!(tile instanceof TileJukeBox)) return null;
                return new ContainerJukeBox(player.inventory,
                        (TileJukeBox) tile);
            default:
                return null;
        }
    }

}

File: src/main/scala/net/cazzar/mods/jukeboxreloaded/network/gui/GuiHandler.scala
package net.cazzar.mods.jukeboxreloaded.network.gui

import net.cazzar.mods.jukeboxreloaded.blocks.tileentity.TileJukebox
import net.cazzar.mods.jukeboxreloaded.network.gui.client.GuiJukebox
import net.cazzar.mods.jukeboxreloaded.network.gui.server.ContainerJukebox
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.BlockPos
import net.minecraft.world.World
import net.minecraftforge.fml.common.network.IGuiHandler
import net.cazzar.mods.jukeboxreloaded.Util._

object GuiHandler extends IGuiHandler{
  final val JUKEBOX = 0

  override def getServerGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef = ID match {
    case JUKEBOX => new ContainerJukebox(player.inventory, new BlockPos(x, y, z).getTileEntityChecked[TileJukebox](world).get)
  }

  override def getClientGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef = ID match {
    case JUKEBOX => new GuiJukebox(player, new BlockPos(x, y, z).getTileEntityChecked[TileJukebox](world).get)
  }
}
